hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/HoodieLogFileCommand.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.cli.commands;
+
+import com.beust.jcommander.internal.Maps;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.uber.hoodie.cli.HoodieCLI;
+import com.uber.hoodie.cli.HoodiePrintHelper;
+import com.uber.hoodie.common.model.HoodieLogFile;
+import com.uber.hoodie.common.model.HoodieRecord;
+import com.uber.hoodie.common.model.HoodieRecordPayload;
+import com.uber.hoodie.common.table.log.HoodieCompactedLogRecordScanner;
+import com.uber.hoodie.common.table.log.HoodieLogFormat;
+import com.uber.hoodie.common.table.log.block.HoodieAvroDataBlock;
+import com.uber.hoodie.common.table.log.block.HoodieCorruptBlock;
+import com.uber.hoodie.common.table.log.block.HoodieLogBlock;
+import com.uber.hoodie.common.util.FSUtils;
+import com.uber.hoodie.config.HoodieCompactionConfig;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.status.api.v1.NotFoundException;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliAvailabilityIndicator;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.stereotype.Component;
+import parquet.avro.AvroSchemaConverter;
+import parquet.schema.MessageType;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+public class HoodieLogFileCommand implements CommandMarker {
+
+  @CliAvailabilityIndicator({"show logfiles"})
+  public boolean isShowArchivedLogFileAvailable() {
+    return HoodieCLI.tableMetadata != null;
+  }
+
+  @CliCommand(value = "show logfile commits", help = "Read commits from log files")
+  public String showLogFileCommits(
+      @CliOption(key = "logFilePathPattern", mandatory = true, help = "Fully qualified path for the log file")
+      final String logFilePathPattern) throws IOException {
+
+
+    List<String> logFilePaths = Arrays.stream(FSUtils.getFs(HoodieCLI.tableMetadata.getBasePath(),
```

Comment:
can we just use the `FileSystemView` or `TableMetaClient` api's here and not glob files directly...I'd like to encourage that route going forward.

Reply:
we can take the fileID and baseCommit here to lead the log files? instead of a glob pattern? If its too hard, happy to concede

Reply:
To populate TableFileSystemView, we need to provide some filestatuses. Also, TableFileSystemView works on partitionPaths/commitTimes which give back multiple file ids. We can think of how to improve the API but for this use case it seems like more change, WDYT ?